zeroex/orderwatch/order_watcher.go

```diff
@@ -56,30 +65,46 @@ type Watcher struct {
 	orderValidator             *ordervalidator.OrderValidator
 	wasStartedOnce             bool
 	mu                         sync.Mutex
+	maxExpirationTime          *big.Int
+	maxOrders                  int
+}
+
+type Config struct {
+	MeshDB           *meshdb.MeshDB
+	BlockWatcher     *blockwatch.Watcher
+	OrderValidator   *ordervalidator.OrderValidator
+	NetworkID        int
+	ExpirationBuffer time.Duration
+	MaxOrders        int
 }
 
 // New instantiates a new order watcher
-func New(meshDB *meshdb.MeshDB, blockWatcher *blockwatch.Watcher, orderValidator *ordervalidator.OrderValidator, networkID int, expirationBuffer time.Duration) (*Watcher, error) {
+func New(config Config) (*Watcher, error) {
 	decoder, err := decoder.New()
 	if err != nil {
 		return nil, err
 	}
 	assetDataDecoder := zeroex.NewAssetDataDecoder()
-	contractAddresses, err := ethereum.GetContractAddressesForNetworkID(networkID)
+	contractAddresses, err := ethereum.GetContractAddressesForNetworkID(config.NetworkID)
 	if err != nil {
 		return nil, err
 	}
 
+	if config.MaxOrders == 0 {
+		config.MaxOrders = defaultMaxOrders
+	}
```

Comment:
This feels like the wrong place to do this. A default should be  set after parsing the configs no? It should be set before any part of the code could potentially use it.

Reply:
I only added it here to make it more convenient to write tests. Since tests don't go through the initialization process in the `core` package, they wouldn't have the defaults applied. That said, I'd be okay with removing it and changing our tests to always specify `MaxOrders`. What do you think?

Reply:
After thinking about it some more I'd prefer to keep it the way it is. It is even more strange to rely on a completely separate package (`core`) and separate code path to apply the correct defaults for `OrderWatcher`. There's a huge potential footgun here where if `MaxOrders` is set to 0 a lot of things will break.

Reply:
Let's throw on 0.